### DISCUSSION TITLE
Portable/Self-Contained Mode

### DISCUSSION THREAD

**Discussion by AsHumanAsICanBe** [ppy/osu#31213] [Ideas]
I'm not a very skilled programmer, and I'm kind of busy right now, so if I can implement this myself, I could only do it as a summer project or something. Whatever, onto the actual subject. I just have a thing for portable, self-contained software, I don't know why, but I think anybody can agree it would be really cool to have an entire osu installation and all your data on a SD card or USB thumb drive. It really shouldn't be hard to implement, just have the data folder be a parent or child folder of the actual game files (the ones in Appdata/Local in windows), and code something in that would make the data folder relative, easy enough, especially for someone who has contributed before and has a good understanding of the codebase and how it is organized. I would really like to see this feature implemented. Cheers!

**Comment by cihe13375** [comment]
The data folder location is already customizable and can be copied/moved to other PCs. For the main installation, idk about windows or mac, but at least the Linux appimage is portable by nature (it can't even be "installed")

**Reply by AsHumanAsICanBe** [reply]
I know that the data folder location can be copied, didn't know linux was portable (that's great btw, I love linux), but windows still isn't. Portable means that you don't have to manually copy anything to run it with the current data, with your understanding of portable almost every single software is portable. I'm not looking to transfer my current data to another PC, I am looking for a completely portable app that runs on any machine (with the correct OS, that is) through a USB without needing to copy anything.

**Comment by trustytrojan** [comment]
this is a good idea but definitely not something the devteam needs to worry about. since i run lazer on both linux and windows on the same pc, i keep my osu data on a separate data (ntfs) partition, and use symbolic links from the usual linux (`~/.local/share/osu`) and windows (`%AppData%\osu`) directories to the osu folder on my separate partition. not hard to do with a little research and command prompt knowledge, and works flawlessly on both os'es.

**Reply by AsHumanAsICanBe** [reply]
Yeah, I know it's not that important but I just posted this in the hopes of somebody who has a good understanding of the codebase and some free time finds this and makes a pull request.

**Comment by peppy** [comment]
Everything already supports this at a framework and installer level. We just need to add some local support to this for the game side as far as I recall. Which is to say it can happen someday, the effort just needs to be put in to make that happen.
